network/alsp/manager/manager_test.go

```diff
@@ -266,6 +267,98 @@ func TestHandleReportedMisbehavior_And_DisallowListing_Integration(t *testing.T)
+
+	// create 1 victim node, 1 honest node and a node for each slashing violation
+	ids, nodes, _ := testutils.LibP2PNodeForMiddlewareFixture(t, 7) // creates 7 nodes (1 victim, 1 honest, 5 spammer nodes one for each slashing violation).
+	mws, _ := testutils.MiddlewareFixtures(t, ids, nodes, testutils.MiddlewareConfigFixture(t), mocknetwork.NewViolationsConsumer(t))
+	networkCfg := testutils.NetworkConfigFixture(t, *ids[0], ids, mws[0], p2p.WithAlspConfig(managerCfgFixture(t)))
+	victimNetwork, err := p2p.NewNetwork(networkCfg)
+	require.NoError(t, err)
+
+	// create slashing violations consumer with victim node network providing the network.MisbehaviorReportConsumer interface
+	violationsConsumer := slashing.NewSlashingViolationsConsumer(unittest.Logger(), metrics.NewNoopCollector(), victimNetwork)
+	mws[0].SetSlashingViolationsConsumer(violationsConsumer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
+	testutils.StartNodesAndNetworks(signalerCtx, t, nodes, []network.Network{victimNetwork}, 100*time.Millisecond)
+	defer testutils.StopComponents[p2p.LibP2PNode](t, nodes, 100*time.Millisecond)
+	defer cancel()
+
+	p2ptest.LetNodesDiscoverEachOther(t, ctx, nodes, ids)
+	// initially victim and misbehaving nodes should be able to connect to each other.
+	p2ptest.TryConnectionAndEnsureConnected(t, ctx, nodes)
+
+	// each slashing violation func is mapped to a violation with the identity of one of the misbehaving nodes
+	// index of the victim node in the nodes slice.
+	victimIndex := 0
+	honestNodeIndex := 1
+	invalidMessageIndex := 2
+	senderEjectedIndex := 3
+	unauthorizedUnicastOnChannelIndex := 4
+	unauthorizedPublishOnChannelIndex := 5
+	unknownMsgTypeIndex := 6
+	slashingViolationTestCases := []struct {
+		violationsConsumerFunc func(violation *network.Violation)
+		violation              *network.Violation
+	}{
+		{violationsConsumer.OnUnAuthorizedSenderError, &network.Violation{Identity: ids[invalidMessageIndex]}},
+		{violationsConsumer.OnSenderEjectedError, &network.Violation{Identity: ids[senderEjectedIndex]}},
+		{violationsConsumer.OnUnauthorizedUnicastOnChannel, &network.Violation{Identity: ids[unauthorizedUnicastOnChannelIndex]}},
+		{violationsConsumer.OnUnauthorizedPublishOnChannel, &network.Violation{Identity: ids[unauthorizedPublishOnChannelIndex]}},
+		{violationsConsumer.OnUnknownMsgTypeError, &network.Violation{Identity: ids[unknownMsgTypeIndex]}},
+	}
+
+	violationsWg := sync.WaitGroup{}
+	violationCount := 120
+	for _, testCase := range slashingViolationTestCases {
+		for i := 0; i < violationCount; i++ {
+			testCase := testCase
+			violationsWg.Add(1)
+			go func() {
+				defer violationsWg.Done()
+				testCase.violationsConsumerFunc(testCase.violation)
+			}()
+		}
+	}
+	unittest.RequireReturnsBefore(t, violationsWg.Wait, 100*time.Millisecond, "slashing violations not reported in time")
```

Comment:
This is done async so that we can ensure all async components under the hood are working as expected (queue, locks, workers etc). This will not cause flakiness as there are no network operations happening, it's just building up a queue of violations that will eventually be processed.